packages/@uppy/dashboard/src/GlobalSearch.browser.test.ts

```diff
@@ -0,0 +1,540 @@
+      height: 500,
+    })
+    .use(RemoteSources, {
+      companionUrl: 'http://companion.test',
+      sources,
+    })
+}
+
+// Removed shared beforeEach initialization. Each test initializes its own Uppy instance.
+
+afterEach(async () => {
+  if (!uppy) return
+
+  const dashboard = uppy.getPlugin('Dashboard') as Dashboard<any, any>
+  dashboard?.hideAllPanels()
+
+  const panelSelector = '[data-uppy-panelType="PickerPanel"]'
+  if (document.querySelector(panelSelector)) {
+    await expect.poll(() => document.querySelector(panelSelector)).toBeNull()
+  }
+
+  uppy.destroy()
+  uppy = undefined
+})
+
+describe('ProviderView Search E2E', () => {
+  test('Search for nested file in Dropbox and verify results', async () => {
+    uppy = initializeUppy(['Dropbox'])
+    await expect
+      .element(page.getByRole('presentation').getByText('Dropbox'))
+      .toBeVisible()
+    await page.getByRole('tab', { name: 'Dropbox' }).click()
+
+    await expect
+      .element(page.getByRole('heading', { name: /import from dropbox/i }))
+      .toBeVisible()
+    const panel = page.getByRole('tabpanel')
+    await expect.element(panel.getByText('test-user@example.com')).toBeVisible()
+
+    const searchInput = document.querySelector(
+      '.uppy-ProviderBrowser-searchFilterInput',
+    ) as HTMLInputElement
+    expect(searchInput).toBeDefined()
+
+    await userEvent.type(searchInput, 'target')
+    // Auto-wait for a specific search result to appear instead of sleeping
+    await expect
+      .element(page.getByRole('button', { name: 'target.pdf', exact: true }))
+      .toBeVisible()
+
+    const targetPdfItem = await page.getByRole('button', {
+      name: 'target.pdf',
+      exact: true,
+    })
+    expect(targetPdfItem).toBeTruthy()
+  })
+
+  test('Search deep folder -> open it -> click ancestor breadcrumb and navigate correctly', async () => {
+    uppy = initializeUppy(['Dropbox'])
+
+    await expect
+      .element(page.getByRole('presentation').getByText('Dropbox'))
+      .toBeVisible()
+    await page.getByRole('tab', { name: 'Dropbox' }).click()
+
+    await expect
+      .element(page.getByRole('heading', { name: /import from dropbox/i }))
+      .toBeVisible()
+    const panel = page.getByRole('tabpanel')
+    await expect.element(panel.getByText('test-user@example.com')).toBeVisible()
+
+    await expect
+      .element(page.getByRole('button', { name: 'first' }))
+      .toBeVisible()
+
+    const searchInput = document.querySelector(
+      '.uppy-ProviderBrowser-searchFilterInput',
+    ) as HTMLInputElement
+
+    await userEvent.clear(searchInput)
+    await userEvent.type(searchInput, 'second')
+
+    await expect
+      .element(page.getByRole('button', { name: 'second' }))
+      .toBeVisible()
+
+    const secondFolder = await page.getByRole('button', { name: 'second' })
+
+    await secondFolder.click()
+
+    await expect
+      .element(
+        page.getByRole('checkbox', { name: 'deep-file.txt', exact: true }),
```

Comment:
In Normal mode (`ListItem.tsx`), folders are rendered as buttons, whereas files are rendered as checkboxes with a corresponding `<label>`.
In Search mode (`SearchListItem.tsx`), both files and folders are rendered as buttons.

Because of this, in Normal mode, when checking whether a file exists, I need to use:

```typescript
await expect.element(page.getByRole('button', { name:'nested-target.pdf', exact: true }))
```

whereas, in Search mode, I need to scope the query to the `checkbox` role instead when searching for a file

Reply:
maybe add exactly that as a comment :)